pytensor/tensor/random/rewriting/basic.py

```diff
@@ -189,10 +183,17 @@ def local_dimshuffle_rv_lift(fgraph, node):
     if config.compute_test_value != "off":
         compute_test_value(new_node)
 
-    out = new_node.outputs[1]
-    if base_rv.name:
-        out.name = f"{base_rv.name}_lifted"
-    return [out]
+    new_next_rng, new_rv = new_node.outputs
+
+    if rv.name:
+        new_rv.name = f"{rv.name}_lifted"
+
+    # We replace uses of the dimshuffled RV by the new RV
+    # And uses of the old RNG update by the new RNG update
+    return {
```

Comment:
I'm not familiar with this type of output. I've always seen lists as return types. What happens with dictionaries? Are they taken as a sort of `updates`?

Reply:
A dictionary allows you to replace any keys by the values. It's an alternative to the default you're used to, which would be something like `dict(zip(node.outputs, new_outputs))`